snmp/tests/test_check.py

```diff
@@ -139,8 +139,34 @@ def test_scalar(aggregator):
 
     # Test service check
     aggregator.assert_service_check("snmp.can_check", status=SnmpCheck.OK, tags=common.CHECK_TAGS, at_least=1)
+    aggregator.assert_metric(
+        'datadog.snmp.submitted_metrics', value=99, metric_type=aggregator.GAUGE, count=1, tags=common.CHECK_TAGS
+    )
 
-    common.assert_common_metrics(aggregator)
+    common.assert_common_metrics(aggregator, tags=common.CHECK_TAGS)
+    aggregator.all_metrics_asserted()
+
+
+def test_submitted_metrics_count(aggregator):
+    instance = common.generate_instance_config(common.SCALAR_OBJECTS)
+    check = common.create_check(instance)
+    check.check(instance)
+
+    for metric in common.SCALAR_OBJECTS:
+        metric_name = "snmp." + (metric.get('name') or metric.get('symbol'))
+        aggregator.assert_metric(metric_name, tags=common.CHECK_TAGS, count=1)
+
+    aggregator.assert_metric('snmp.sysUpTimeInstance', count=1)
+    aggregator.assert_metric(
+        'datadog.snmp.submitted_metrics', value=5.0, metric_type=aggregator.GAUGE, count=1, tags=common.CHECK_TAGS
+    )
+    aggregator.assert_metric(
+        'datadog.snmp.check_duration', metric_type=aggregator.GAUGE, count=1, tags=common.CHECK_TAGS
+    )
+    aggregator.assert_metric(
+        'datadog.snmp.check_interval', metric_type=aggregator.MONOTONIC_COUNT, count=1, tags=common.CHECK_TAGS
+    )
+    common.assert_common_device_metrics(aggregator, tags=common.CHECK_TAGS)
     aggregator.all_metrics_asserted()
```

Comment:
Should we add this?
`aggregator.assert_metrics_using_metadata(get_metadata_metrics(), check_submission_type=True)`

Reply:
@gmarabout Sure, updated here: https://github.com/DataDog/integrations-core/pull/8211/commits/3c8be91b3a5a07f3d2b71e5bc4b1927d2189d537